drjt/MyDate/scripts/mydate.pl

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+use 5.010;
+use warnings;
+use strict;
+
+use File::Basename qw( basename );
+
+use MyDate;
+
+say "Date is ", MyDate::date();
+say "Month is ", MyDate::month();
+say "Year is ", MyDate::year();
+
+MyDate->debug( "Moo" );
+
+sub UNIVERSAL::debug {
+  shift;
+  printf( "%s: %s\n", MyDate::date(), join( ' ', @_ ) );
```

Comment:
You could include ``@_`` in the template and it would automatically be interpolated with spaces, saving the effort of joining it yourself.